server/irmaserver/helpers.go

```diff
@@ -429,3 +504,51 @@ func (s *Server) sessionMiddleware(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r.WithContext(context.WithValue(ctx, "session", session)))
 	})
 }
+
+func (s *Server) authenticationMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		session := r.Context().Value("session").(*session)
+
+		authorization := r.Header.Get(irma.AuthorizationHeader)
+		if authorization == "" && session.version != nil {
```

Comment:
If `version` is nil, then there is no version negotiated yet (i.e. the client is not connected yet). This means we do not know yet whether the app might be a legacy version. When it is a legacy version then we have to let it pass, since legacy apps do not know a authorization header. As soon as the client is connected, version will be set and then this check will never be activated. 

In fact this check is in practice exactly the same to `session.status == server.StatusInitialized`. However, if I do this it means that the behavior of the irma server will change when a legacy app would do its first call to another endpoint than `/`. Normally this wouldn't happen, so I think it would be no problem. However, the legacy API does change from behaviour then. I don't know whether that is a problem...